site/viamguides/src/ncnn-edge-ai/ncnn-edge-ai.md

```diff
@@ -0,0 +1,246 @@
+   ![viam agent setup script in SSH session](./assets/viam-agent-setup.png)
+1. The setup page will indicate when the machine is successfully connected.
+   ![machine connected successfully](./assets/machine-connected.png)
+
+<!-- ------------------------ -->
+## Configure your webcam
+Duration: 5
+
+1. In [the Viam app](https://app.viam.com/fleet/locations) under the **CONFIGURE** tab, click the **+** icon in the left-hand menu and select **Component or service**.
+   ![select component or service](./assets/add-component.png)
+1. Select `camera`, and find the `webcam` module. This adds the module for working with a USB webcam. Leave the default name `camera-1` for now.
+   ![add camera component](./assets/name-camera.png)
+1. Notice adding this component adds the webcam hardware component called `camera-1`. From the **Configure** section of the panel, switch to the JSON configuration mode using the `{}` icon in the top-right:
+   ![new camera component in configuration](./assets/new-camera-component.png)
+   ![switch configuration mode](./assets/camera-advanced-config.png)
+1. In this mode, configure your component with the following JSON in the **CONFIGURE** field. This will tell the component to look for the default camera connected to the device, which will be the USB webcam.
+   ```json
+   {
+       "video_path": ""
+   }
+   ```
+   ![set the blank video path](./assets/camera-component-config.png)
+1. Click **Save** in the top right. This may take a moment to apply your configuration changes.
+
+   > aside positive
+   > If multiple webcams are connected to your Pi, you'll need to specify the correct video path. To list available camera devices, you can use the [`rand:find-webcams:webcam-discovery`](https://app.viam.com/module/rand/find-webcams)
+
+1. At the bottom of the `camera-1` panel, expand the **TEST** section to ensure you have configured the camera properly and see a video feed.
+   ![Test camera feed](./assets/camera-test.png)
+
+<!-- ------------------------ -->
+## Add and configure the ncnn ML model
+Duration: 10
+
+Now, let's add the ncnn ML model service to our machine. This will let us run optimized neural networks for computer vision.
+
+### Add the ML model service
+
+1. In [the Viam app](https://app.viam.com/fleet/locations) under the **CONFIGURE** tab, click the **+** icon in the left-hand menu and select **Service**.
+1. Select `ML Model`, and find the `hipsterbrown:mlmodel:ncnn` module. This adds the module for running optimized neural networks with ncnn. Leave the default name `mlmodel-1` for now.
+   ![find ncnn ML model service](./assets/find-mlmodel.png)
+   ![add ncnn ML model service](./assets/add-mlmodel.png)
+1. Notice adding this service adds the ML model service called `mlmodel-1`.
+   ![new ML model service](./assets/new-mlmodel-service.png)
+1. In the new `mlmodel-1` panel, configure your service with the following JSON in the **Attributes** section:
```

Comment:
If there's not a physical section labeled **Attributes** in the UI, wouldn't call it that with bold. Sometimes we refer to attributes, dependencies, configuration, **CONFIGURE** panel all over (README, docs, codelabs), and it can be confusing if someone is looking for that UI element. If you'd like to call it "attributes", perhaps lower case and unbold.